File: 63_Name/python/name.py
"""
NAME

simple string manipulations on the user's name

Ported by Dave LeCompte
"""


def is_yes_ish(answer: str) -> bool:
    cleaned = answer.strip().upper()
    if cleaned in ["Y", "YES"]:
        return True
    return False


def main() -> None:
    print(" " * 34 + "NAME")
    print(" " * 15 + "CREATIVE COMPUTING  MORRISTOWN, NEW JERSEY\n\n\n")
    print("HELLO.")
    print("MY NAME iS CREATIVE COMPUTER.")
    name = input("WHAT'S YOUR NAME (FIRST AND LAST)?")
    print()
    name_as_list = list(name)
    reversed_name = "".join(name_as_list[::-1])
    print(f"THANK YOU, {reversed_name}.\n")
    print("OOPS!  I GUESS I GOT IT BACKWARDS.  A SMART")
    print("COMPUTER LIKE ME SHOULDN'T MAKE A MISTAKE LIKE THAT!\n\n")
    print("BUT I JUST NOTICED YOUR LETTERS ARE OUT OF ORDER.")

    sorted_name = "".join(sorted(name_as_list))
    print(f"LET'S PUT THEM IN ORDER LIKE THIS: {sorted_name}\n\n")

    print("DON'T YOU LIKE THAT BETTER?")
    like_answer = input()
    print()
    if is_yes_ish(like_answer):
        print("I KNEW YOU'D AGREE!!")
    else:
        print("I'M SORRY YOU DON'T LIKE IT THAT WAY.")
    print()
    print(f"I REALLY ENJOYED MEETING YOU, {name}.")
    print("HAVE A NICE DAY!")


if __name__ == "__main__":
    main()

File: 63_Name/ruby/name.rb
def is_yes_ish answer
    cleaned = answer.upcase
    return true if ["Y", "YES"].include? cleaned
    return false
end

def main
    puts " " * 34 + "NAME"

    puts "HELLO."
    puts "MY NAME iS COMPUTER."
    print "WHAT'S YOUR NAME (FIRST AND LAST)? "
    name = gets.chomp!
    puts ""
    name_as_list = name.split("")
    reversed_name = name_as_list.reverse.join("")

    puts "THANK YOU, #{reversed_name}.\n"
    puts "OOPS!  I GUESS I GOT IT BACKWARDS.  A SMART"
    puts "COMPUTER LIKE ME SHOULDN'T MAKE A MISTAKE LIKE THAT!\n\n"
    puts "BUT I JUST NOTICED YOUR LETTERS ARE OUT OF ORDER."

    sorted_name = name_as_list.sort.join("")
    puts "LET'S PUT THEM IN ORDER LIKE THIS: #{sorted_name}\n\n"
    print "DON'T YOU LIKE THAT BETTER? "
    like_answer = gets.chomp!
    puts
    if is_yes_ish(like_answer)
        puts "I KNEW YOU'D AGREE!!"
    else
        puts "I'M SORRY YOU DON'T LIKE IT THAT WAY."
    end

    puts ""
    puts "I REALLY ENJOYED MEETING YOU, #{name}."
    puts "HAVE A NICE DAY!"
end


if __FILE__ == $0
    main
end